exchanges/orderbook/orderbook.go

```diff
@@ -13,6 +13,40 @@ import (
 	"github.com/thrasher-corp/gocryptotrader/log"
 )
 
+// GetAveragePrice finds the average buy or sell price of a specified amount.
+// It finds the nominal amount spent on the total purchase or sell and uses it
+// to find the average price for an individual unit bought or sold
+func (b *Base) GetAveragePrice(buy bool, amount float64) (float64, error) {
+	var aggNominalAmount, remainingAmount float64
+	if buy {
+		aggNominalAmount, remainingAmount = b.Asks.FindNominalAmount(amount)
+	} else {
+		aggNominalAmount, remainingAmount = b.Bids.FindNominalAmount(amount)
+	}
+	if remainingAmount != 0 {
+		return 0, fmt.Errorf("%w for %v on exchange %v to support a buy amount of %v", errNotEnoughLiquidity, b.Pair, b.Exchange, amount)
+	}
+	return aggNominalAmount / amount, nil
+}
+
+// FindNominalAmount finds the nominal amount spent in terms of the quote
+// If the orderbook doesn't have enough liquidity it returns a non zero
+// remaining amount value
+func (i Items) FindNominalAmount(amount float64) (aggNominalAmount, remainingAmount float64) {
```

Comment:
Please follow github advice and update all other Items receiver function variables

Reply:
Have changed the (i Items) to (elem Items) now 🎉